vault/core.go

```diff
@@ -651,12 +649,20 @@ type Core struct {
 	// Whether we use a single global memdb instance for identity; see
 	// commentary below.
 	unsafeCrossNamespaceIdentity bool
+
+	// Core invalidation tracker handles dispatching invalidations and
+	// refreshing the Core-adjacent caches afterwards.
+	invalidations *invalidationManager
 }
 
 // c.stateLock needs to be held in read mode before calling this function.
 func (c *Core) HAState() consts.HAState {
 	switch {
 	case c.standby.Load():
+		if c.StandbyReadsEnabled() {
+			return consts.PerfStandby
+		}
```

Comment:
I think we initially did want to distinguish between "normal" and performance standby (as performance standby / standby reads is not "paywalled")

I'm fine with adding this distinction, since we added a flag to explicitly disable standby reads.